cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1562,6 +1577,10 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 		}
 	}
 
+	if err := machine.degradedState.serializeTo("unlocked.json"); err != nil {
```

Comment:
I think it might be slightly cleaner to add a writeUnlockState to to machine and have the if about writing degraded.json there and update the step comment